aten/src/ATen/TensorUtils.cpp

```diff
@@ -367,19 +367,26 @@ inline static std::optional<ResultVec> computeStride_impl(
   // numel in current chunk
   Numel tensor_numel = 1;
   Numel view_numel = 1;
+
+ // The usages of TORCH_GUARD_OR_TRUE/TORCH_GUARD_OR_FALSE below could result in returning std::nullopt which have an effect of falling
+ // back to a clone when unbacked presented. But it will not result in returning different or wrong results.
   for (int64_t tensor_d = oldshape.size() - 1; tensor_d >= 0; tensor_d--) {
     tensor_numel *= oldshape[tensor_d];
     // if end of tensor size chunk, check view
     if ((tensor_d == 0) ||
-        (TORCH_GUARD_SIZE_OBLIVIOUS(sym_ne(oldshape[tensor_d - 1], 1)) &&
-         TORCH_GUARD_SIZE_OBLIVIOUS(sym_ne(oldstride[tensor_d - 1], tensor_numel * chunk_base_stride)))) {
+        (TORCH_GUARD_OR_TRUE(sym_ne(oldshape[tensor_d - 1], 1)) &&
+        TORCH_GUARD_OR_TRUE(sym_ne(oldstride[tensor_d - 1], tensor_numel * chunk_base_stride)))) {
+     // We want to accumulate stuff in view_numel until view_numel == tensor_numel, if we do not know yet if they are we keep
+     // accumulating. if view_numel<tensor_numel view_numel==tensor_numel would fail also so better to look ahead.
+     // we use TORCH_GUARD_OR_FALSE when comparing newshape[view_d] ==1 because if we know view_numel<tensor_numel is false.
+     // we want to stop. Unless we know for sure newshape[view_d]==1 in that case we would stop in the next iteration anyway.
```

Comment:
I find this entire comment very hard to follow. It might be easier to do some case analysis with some concrete examples.